File: tests/python/error_test_old.py
import os,sys
sys.path.insert(0,os.path.join(os.path.dirname(os.path.abspath(__file__)),"../../../chargebee-python"))
#print os.path.join(os.path.dirname(os.path.abspath(__file__)),"../../python")
import chargebee
import json
from chargebee.main import Environment
from chargebee import APIError
Environment.chargebee_domain="localcb.in:8080"
chargebee.ChargeBee.verify_ca_certs=False

##Copy code from api docs
chargebee.configure("test___dev__NqcddV3mrcu9pWqcyacuq0twyYyXXPrVlcuj","mannar-test")

def print_error(e):
    print "OLD ERRORS"
    print e.error_code
    print e.http_code
    print e.message
    print e.param
    print "NEW ERRORS"
    print e.json_obj["type"]
    print e.json_obj["api_error_code"]
    print e.json_obj.get("param") 
    print e.json_obj["message"]
    #print json.dumps(e.json_obj)

def create_sub():
  try :
    result = chargebee.Subscription.create({
      "plan_id" : "basic", 
      "coupon" : "plan_only_coupon",
      "customer" : {
        # "email" : "johnasdsaduser.com", 
         "first_name" : "John", 
         "last_name" : "Doe", 
         "phone" : "+1-949-999-9999"
      }, 
      "billing_address" : {
         "first_name" : "John", 
         "last_name" : "Doe", 
         "line1" : "PO Box 9999", 
         "city" : "Walnut", 
         "state" : "CA", 
         "zip" : "91789", 
         "country" : "US"
      }, 
      "addons" : [{
        "id" : "ssl"
     }]
    })
    print result
  except APIError as e:
    print_error(e)

def update_subscription():
  try:
    result = chargebee.Subscription.update("__dev__XpbG8HXOpwIWifu", {
        "plan_id" : "basic", 
        "billing_address" : {
            "first_name" : "John", 
            "last_name" : "Doe", 
            "line1" : "PO Box 9999", 
            "city" : "Walnut", 
            "state" : "CA", 
            "zip" : "600059",
            "country" : "US"
       }
    })
  except APIError as e:
    print_error(e)

def hosted_pages():
  try:
    result = chargebee.HostedPage.checkout_new({
       "subscription" : {
           "plan_id" : "basic",
           "coupon" : "max_redem"
        }, 
        "customer" : {
           "email" : "john@user.com", 
           "first_name" : "John", 
           "last_name" : "Doe", 
           "phone" : "+1-949-999-9999"
        }, 
        "addons" : [{
           "id" : "ssl"
        }]
    })
    print result
  except APIError as e:
    print_error(e)

def create_invoice_for_charge():
  try:
     result = chargebee.Invoice.charge({
        "customer_id" : "asactive_direct", 
        "amount" : "1000", 
        "description" : "Support charge"
     })
     print result
  except APIError as e:
     print_error(e)

def create_invoice_for_addon():
  try:
    result = chargebee.Invoice.charge_addon({
          "subscription_id" : "active_direct",
          "addon_id" : "day_pass",
          "addon_quantity" : "qqwe12"
    })
    print result
  except APIError as e:
    print_error(e)

def list_events():
  try:
     result = chargebee.Event.list({
           "start_time" : 1391299200,
           "end_time" : 1356998400 ,
      })
  except APIError as e:
     print_error(e)


#create_sub()
#update_subscription()
#hosted_pages()
#create_invoice_for_charge()
#create_invoice_for_addon()
list_events()

File: tests/php/error_test_old.php
<?php
require '../../../chargebee-php/lib/ChargeBee.php';
ChargeBee_Environment::$scheme = "http";
ChargeBee_Environment::$chargebeeDomain = "localcb.in:8080";

ChargeBee_Environment::configure("mannar-test","test___dev__NqcddV3mrcu9pWqcyacuq0twyYyXXPrVlcuj");

function printError($e) {
   echo "OLD ERRORS\n";
   $errorJSON = $e->getJSONObject();
   echo $errorJSON["error_code"] . "\n";
   echo $errorJSON["error_msg"] . "\n";
   echo $errorJSON["http_status_code"] . "\n";
   if( isset($errorJSON["error_param"])) {
     echo $errorJSON["error_param"] . "\n"; 
   }
   echo "NEW ERRORS\n";
   echo $errorJSON["type"] . "\n"; 
   echo $errorJSON["api_error_code"] . "\n";
   echo $errorJSON["message"] . "\n";
   echo $errorJSON["http_status_code"] . "\n";
   if( isset($errorJSON["param"]) ) {
     echo $errorJSON["param"] . "\n";
   }
   echo json_encode($errorJSON) . "\n"; 
}

function createSub() {
try{
 $result = ChargeBee_Subscription::create(array(
  "planId" => "flat_fee", 
 "plan_quantity" => "2",
//  "coupon" => "plan_only_coupon",
  "customer" => array(
    "email" => "john@user.com", 
//    "firstName" => "John asdasdabk asjbdakjshbd asjbdkjahs akjbsdsak aBDBSFLDSBF  asdbflas asdasda asdhjasbkf asdhfjbjkhadsas asdasd asdahbdaks askdbsakjd akjsdnsjnad asjdnsad dadnsadnbaloiwerywer asbajkbauiewr asbbasoweru", 
    "lastName" => "Doe", 
    "phone" => "+1-949-999-9999",
  ), 
  "billingAddress" => array(
    "firstName" => "John", 
    "lastName" => "Doe", 
    "line1" => "PO Box 9999", 
    "city" => "Walnut", 
    "state" => "CA", 
    "zip" => "91789", 
    "country" => "US"
  ), 
/*  "card" => array( 
    "number" => "4111111111111111",
    "expiry_month" => "12",
    "expiry_year" => "2014"
  )*/
//  "addons" => array(array(
//    "id" => "ssl"
//  )) 
  ));
 echo "success\n";
} catch(ChargeBee_APIError $e) {
  printError($e);
}

}

function updateSubscription() {
 try {
  $result = ChargeBee_Subscription::update("__dev__XpbG8HXOpwEt9RQ", array(
           "planId" => "professional", 
         //  "planQuantity" => "3",
          "coupon" => "plan_coupon",
           "billingAddress" => array(
                   "firstName" => "John", 
                   "lastName" => "Doe", 
                   "line1" => "PO Box 9999", 
                   "city" => "Walnut", 
                   "state" => "CA", 
                   "zip" => "91789", 
                   "country" => "US"
             ), 
             "addons" => array(array(
                   "id" => "day_pass",
                   "quantity" => "2"
             ))
   ));
   echo "success\n";
 } catch(ChargeBee_APIError $e) {
   printError($e);
 }
}

function cancelSub() {
try {
  $result = ChargeBee_Subscription::cancel("future_billing");
  echo "api call success";
} catch(ChargeBee_APIError $e) {
 printError($e);
}
}
function hostedPages() {
 try {
  $result = ChargeBee_HostedPage::checkoutNew(array(
  "subscription" => array(
    "planId" => "basic",
    "coupon" => "plan_coupon"
  ), 
  "customer" => array(
    "email" => "john@user.com", 
    "firstName" => "John", 
    "lastName" => "Doe", 
    "phone" => "+1-949-999-9999"
  ) 
//  "addons" => array(array(
//    "id" => "sslaaa"
//    "quantity" => "2"
//  ))
   ));
  echo "success\n";
 } catch(ChargeBee_APIError $e) {
   printError($e); 
 }
}


function checkoutExisting() {
try {
 $result = ChargeBee_HostedPage::checkoutExisting(array(
  "subscription" => array(
    "id" => "__dev__XpbG8HXOpwEt9RQ", 
    "planId" => "flat_fee",
    "planQuantity" => "3"
  )
 /* "addons" => array(array(
    "id" => "ssl"
  )) */
 ));
 echo "success"; 
} catch(ChargeBee_APIError $e) {
 printError($e);
}
}

function createInvoiceForCharge() {
 try {
  $result = ChargeBee_Invoice::charge(array(
       "subscriptionId" => "active_direct", 
       "amount" => "1000asdsa", 
       "description" => "Support charge"));
    echo "success";
  } catch(ChargeBee_APIError $e) {
    printError($e);
  }
}

function createInvoiceForAddon() {
 try {
  $result = ChargeBee_Invoice::chargeAddon(
               array("subscriptionId" => "active_direct",
                     "addonId" => "day_pass",
                     "addonQuantity" => "asd2"));
  echo "success";                             
 } catch(ChargeBee_APIError $e) {
  printError($e);
 }
}

function listEvents() {
try {
 $all = ChargeBee_Event::all(array(
       "limit" => 5, 
       "startTime" => 1388534400, 
       "endTime" => 1325376000));
  echo "success";
} catch( ChargeBee_APIError $e) {
  printError($e);
}
}


//createSub()
//cancelSub()
//updateSubscription()
//hostedPages()
//checkoutExisting()
//createInvoiceForCharge()
//createInvoiceForAddon()
listEvents()
?>
